version: '3'

volumes:
  postgres_data:

services:

  db:
    image: postgres:13.0-alpine
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    env_file:
      - ./.env
    ports:
      - "5432:5432"

  test_db:
    image: postgres:13.0-alpine
    environment:
      POSTGRES_PASSWORD: ${TEST_POSTGRES_PASSWORD}
      POSTGRES_USER: ${TEST_POSTGRES_USER}
      POSTGRES_DB: ${TEST_POSTGRES_DB}
    ports:
      - "5433:5432"

  redis:
    image: redis:6.0.9
    restart: on-failure
    ports:
      - "6379:6379"

  auth_api:
    build: auth_api
    env_file:
      - .env
    expose:
      - "8000"
    depends_on:
      - redis
      - db

  nginx:
    container_name: nginx-container
    image: nginx:latest
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf:ro
      - ./configs:/etc/nginx/conf.d:ro
      - ./data:/data/:ro
    depends_on:
      - db
      - redis
      - auth_api
    ports:
      - "8000:80"

  jaeger:
    image: jaegertracing/all-in-one:latest
    ports:
      - "6831:6831/udp"
      - "16686:16686"
